crypto/ec/ecdsa_ossl.c

```diff
@@ -97,6 +102,11 @@ int ossl_ecdsa_deterministic_sign(const unsigned char *dgst, int dlen,
     BIGNUM *kinv = NULL, *r = NULL;
     int ret = 0;
 
+    if (sig == NULL) {
+        ERR_raise(ERR_LIB_EC, ERR_R_PASSED_NULL_PARAMETER);
+        return 0;
```

Comment:
Why do we return error for deterministic ECDSA and for SM2, but we return ECDSA_size() for ECDSA?

Reply:
Ah, it's handled in the provider code. OK then.